packages/lesswrong/components/Layout.tsx

```diff
@@ -55,13 +55,49 @@ const standaloneNavMenuRouteNames: Record<string,string[]> = {
 
 const styles = theme => ({
   main: {
-    margin: '50px auto 15px auto',
+    paddingTop: 50,
+    paddingBottom: 15,
+    marginLeft: "auto",
+    marginRight: "auto",
+    background: "#f4f4f4",
+    minHeight: "100vh",
+    gridArea: 'main', 
     [theme.breakpoints.down('sm')]: {
-      marginTop: 0,
+      paddingTop: 0,
       paddingLeft: theme.spacing.unit/2,
       paddingRight: theme.spacing.unit/2,
     },
   },
+  gridActivated: {
+    '@supports (grid-template-areas: "title")': {
+      display: 'grid',
+      gridTemplateAreas: `
+        "navSidebar ... main ... sunshine"
+      `,
+      gridTemplateColumns: `
+      minmax(0, 250px)
+      minmax(0, 1fr)
+      minmax(800px, 800px)
+      minmax(0, 1fr)
+      minmax(0, 1fr)
```

Comment:
Currently this doesn't handle "sunshine sidebar is visible/non-visible" properly.

Reply:
It also doesn't quite handle the size of PostItems right – when initially loading a page, it sometimes hasn't loaded post items, and the width collapses to too small.